examples/java/src/test/java/org/apache/beam/examples/WordCountIT.java

```diff
@@ -46,9 +68,121 @@
   public void testE2EWordCount() throws Exception {
     PipelineOptionsFactory.register(WordCountITOptions.class);
     WordCountITOptions options = TestPipeline.testingPipelineOptions().as(WordCountITOptions.class);
-    options.setOutput(Joiner.on("/").join(new String[]{options.getTempRoot(),
-        String.format("WordCountIT-%tF-%<tH-%<tM-%<tS-%<tL", new Date()), "output", "results"}));
+    options.setOutput(
+        Joiner.on("/")
+            .join(
+                new String[] {
+                  options.getTempRoot(),
+                  String.format("WordCountIT-%tF-%<tH-%<tM-%<tS-%<tL", new Date()),
+                  "output",
+                  "results"
+                }));
+    options.setOnSuccessMatcher(new WordCountOnSuccessMatcher(options.getOutput() + "*"));
 
     WordCount.main(TestPipeline.convertToArgs(options));
   }
+
+  /**
+   * Matcher for verifying WordCount output data.
+   */
+  static class WordCountOnSuccessMatcher extends BaseMatcher<PipelineResult>
```

Comment:
Look at TypeSafeMatcher, will get rid of a bunch of boilerplate:
http://hamcrest.org/JavaHamcrest/javadoc/1.3/org/hamcrest/TypeSafeMatcher.html#describeMismatch(java.lang.Object, org.hamcrest.Description)